recipe/meta.yaml

```diff
@@ -19,16 +19,20 @@ outputs:
       run_exports:
         strong:
           - cudatoolkit {{ cuda_compiler_version }}|{{ cuda_compiler_version }}.*
+          - __glibc >=2.17  # [cuda_compiler_version == "11.0"]
     requirements:
       host:
         # Needed to symlink libcuda into sysroot libs.
         - {{ compiler("c") }}
+        - sysroot_linux-64 2.17  # [cuda_compiler_version == "11.0"]
       run:
+        - sysroot_linux-64 2.17  # [cuda_compiler_version == "11.0"]
```

Comment:
If you need glibc >=2.17 at runtime, then we need to make sure the virtual package constraint makes it here.

Reply:
It's not needed at runtime. Only at runtime of downstream packages.

Reply:
Well downstream packages should be built with GLIBC 2.17. So wouldn't we want that constraint to make sure errors are raised if that is not met?

Reply:
@isuruf @jakirkham wrote this

> Just to provide more context, this package acts as a shim to interact with an actual install of nvcc. So all nvcc installs for CUDA 11.0 require GLIBC 2.17+. I suppose this package could be installed in an environment without that, but it wouldn't be of much practical use.

I think GLIBC 2.17+ is needed at runtime according to this statement.

Reply:
This is a shim package. there's no need to have glibc 2.17 for a shim package with just an activation script.

Reply:
Fair but there are two things here. The second is

> I suppose this package could be installed in an environment without that, but it wouldn't be of much practical use.

Why let people install a shim of no use in an env? For the shim to be usable at all for CUDA 11, it needs a machine with GLIBC >=2.17 so it should express that in its constraints.

Reply:
It is usable on a machine without glibc>=2.17 if you use a different compiler like `clang` although there aren't use cases in conda-forge right now.

Reply:
Ah. So this statement

> So all nvcc installs for CUDA 11.0 require GLIBC 2.17+.

Is not true then. Thanks!

Reply:
To run `nvcc`, it is required, but this package does more than that and adds flags to CUDA_HOME, etc.

Reply:
Ahh you mean run clang instead of nvcc. This package is meant for nvcc only as I understand it. It seems like if we want to support running clang instead of nvcc, we should separate the flags from the nvcc stuff and use proper constraints on each one.

Reply:
A little unclear on whether we want this change or not. Went ahead and included it in the latest commit. Though feel free to comment on it if you think changes are needed.

Reply:
I think @beckermr was okay with not having having this change as he marked the conversation as resolved.

Reply:
I don't think it is a great idea to leave this package with formally incorrect requirements so it can be hacked on, but I  also don't really care. I was asking questions to make sure I understood what was going on.

Reply:
Maybe we can comment on the change more directly? It's in the latest commit below.